### DISCUSSION TITLE
How to enable the mysql DB with GPTcache?

### DISCUSSION THREAD

**Discussion by terryweijian** [zilliztech/GPTCache#375] [Q&A] [answered]
Could you please introduce how to setup the GPTcache with mysql database? Or example of using mysql with GPTcache.

suppose there should be a config file to store the mysql connection information like host, port, user, password and etc. 
or if can use the following method?
cache_base = CacheBase('mysql'  host='127.0.0.1', port='3306', user='root', password='XXXX')

Thanks

**Comment by SimFG** [comment, marked as answer]
you can try the following code snippet:

```
from gptcache.embedding import Onnx
from gptcache.manager import manager_factory

onnx = Onnx()
data_manager = manager_factory(
    "mysql,faiss",
    scalar_params={
        "sql_url": "mysql+pymysql://root:123456@127.0.0.1:3306/mysql",
        "table_name": "gptcache",
    },
    vector_params={"dimension": onnx.dimension},
)
```

more examples: https://github.com/zilliztech/GPTCache/blob/main/examples/data_manager/scalar_store.py

**Reply by terryweijian** [reply]
Thanks for help.
I used the following format to connect with mysql server. It works!. (ps, need to manually create a database call "justtest" in mysql if don't want to use the default "mysql" database.

cache_base = CacheBase('mysql', sql_url='mysql+pymysql://root:*****@127.0.0.1:3306/justtest')
vector_base = VectorBase('faiss', dimension=huggingface.dimension)
data_manager = get_data_manager(cache_base, vector_base, max_size=10, clean_size=2)

**Reply by SimFG** [reply]
yes, you are right about the database.

**Reply by Coditas-Vikas-Tiwari** [reply]
Hey I'm usin the same approach but it's not storing any data into my table.

**Reply by SimFG** [reply]
you can show me your demo code

**Reply by prakashpatidarcoindcx** [reply]
Hi I am using this approach here is my code and this is creating all required table starting with gptcache and using cache until process lifespan but the moment I restart the process it does not use cache for all question which is already stored in gptcache_answer.
tables and data is persisted in gptcache table but not sure why it is not able to use cached answer for question which is being asked before process start. here is code base:
self.onnx = Onnx()

        
        self.data_manager = manager_factory(
            "mysql,faiss",
            scalar_params={
                "sql_url": "mysql+pymysql://root:new_password@127.0.0.1:3306/mysql",
                "table_name": "gptcache",
            },
            vector_params={"dimension": self.onnx.dimension},
        )
        cache.init(
            embedding_func=self.onnx.to_embeddings,
            data_manager=self.data_manager,
            similarity_evaluation=SearchDistanceEvaluation(),
            config=Config(similarity_threshold=0.88,auto_flush=5),
            )
        cache.set_openai_key()

Any help to ensure that caching can be used across process lifespan will be really helpful.
Let me know if you need more context here.

**Reply by SimFG** [reply]
please give me the code about the cache usage in the llm. like
```
openai.ChatCompletion.create()
```
